### PR TITLE
VSCE: Add dismiss button to CTA banners

### PR DESCRIPTION
**Pull request by abeatrix**
PR to add dismiss button to CTA banners
Close: https://github.com/sourcegraph/sourcegraph/issues/31089
image

## Test plan

N/A

### CODE REVIEW

**Comment by jjinnii** [comment]
@abeatrix 
I noticed that we've actually changed the styling of the banner on sourcegraph.com (moved the button + copy of the button). Would we be able to quickly pull in these changes as a part of this PR? 😄 

Screen Shot 2022-04-03 at 9 29 56 PM

File: client/vscode/src/webview/search-panel/components/SearchCta.tsx [hunk 1]
```diff
@@ -17,23 +21,44 @@ export const SearchPageCta: React.FunctionComponent<SearchPageCtaProps> = ({
...
-}) => (
-    <div className={classNames(styles.container)}>
-        <div className="mr-md-3 ml-3">
-            <div className="w-50">{icon}</div>
-        </div>
-        <div className={classNames('flex-1 my-md-0 my-2', styles.contentContainer)}>
-            <div className={classNames('mb-1', styles.title)}>
-                <strong>{ctaTitle}</strong>
+}) => {
+    const [show, setShow] = useState(true)
```
Comment (tjkandala) [inline client/vscode/src/webview/search-panel/components/SearchCta.tsx:25]: Should the dismissal be persistent? Or do we like that the banner will show up again?

**Review by tjkandala — APPROVED** [review]

File: client/vscode/src/webview/search-panel/components/SearchCta.tsx (hunk 1 above)
Comment (jjinnii) [reply client/vscode/src/webview/search-panel/components/SearchCta.tsx:25]: @jjinnii I think it should be persistent! I can't think of a logic to re-show the banner after dismissal that wouldn't annoy an user 😄

**Comment by abeatrix** [comment]
The base branch has made some new changes that resulted in lots of conflicts so I have started a new branch to continue this in a new PR: https://github.com/sourcegraph/sourcegraph/pull/33621

I have updated to use the banner from the main app and dismissing the banner will be persistent (store in vs code storage):  
image

**Pull request closed without merging**